src/jobs/assets.yml

```diff
@@ -0,0 +1,25 @@
+description: "Precompile assets with the `assets:precompile` task."
+
+parameters:
+  executor:
+    description: "Executor to use for this job"
+    type: executor
+  before-steps:
+    description: "Steps to run before assets:precompile"
+    type: steps
+    default: []
+
+executor: << parameters.executor >>
+
+steps:
+  - checkout
+  - << parameters.before-steps >>
```

Comment:
Does this work?
I found some code like the following in [GitHub search](https://github.com/search?q=before-steps%3A&type=Code).

```yaml
steps:
  ...
  - steps: << parameters.before-steps >>
  ...
```

How about adding `before-steps` tests to integration test.

Reply:
I implemented this with reference to [circleci/orb-tools](https://github.com/CircleCI-Public/orb-tools-orb/blob/v9.0.0/src/jobs/test-in-builds.yml#L60).

However, I thought it would be better to follow the example in the official documentation, and fixed it.
ref: https://circleci.com/docs/2.0/reusing-config/#steps

Reply:
I don't know how to test the job provided by Orb.

I have not written integration-tests, but checked that the settings in `examples/rspec.yml` works with `medpeer/rails@dev:41b114a`.